src/IECoreMaya/ToMayaPlugConverter.cpp

```diff
@@ -68,12 +71,100 @@ ToMayaPlugConverterPtr ToMayaPlugConverter::create( const IECore::ObjectPtr src
 	return new ToMayaPlugConverter( src );
 }
 
-bool ToMayaPlugConverter::convert( MPlug &plug ) const
+template <typename DataType, typename UnitType>
+MStatus ToMayaPlugConverter::setPlugValue(MPlug &plug, boost::intrusive_ptr<const IECore::TypedData<DataType> > data)
 {
-	ConstObjectPtr toConvert = srcParameter()->getValidatedValue();
+	MStatus s;
+	s = plug.setValue(data->readable());
+	return s;
+}
+
 
+// specializations for Unit Attributes
+namespace IECoreMaya
+{
+
+template <>
+MStatus ToMayaPlugConverter::setPlugValue<float, MTime>(MPlug &plug, boost::intrusive_ptr<const IECore::TypedData<float> > data)
+{
 	MStatus s;
+	MTime t;
+	t.setUnit(MTime::kSeconds);
+	t.setValue(data->readable());
+	s = plug.setValue(t);
+	return s;
+}
+
+template <>
+MStatus ToMayaPlugConverter::setPlugValue<float, MDistance>(MPlug &plug, boost::intrusive_ptr<const IECore::TypedData<float> > data)
+{
+	MStatus s;
+	MDistance t;
+	t.setUnit(MDistance::kCentimeters);
+	t.setValue(data->readable());
+	s = plug.setValue(t);
+	return s;
+}
+
+template <>
+MStatus ToMayaPlugConverter::setPlugValue<float, MAngle>(MPlug &plug, boost::intrusive_ptr<const IECore::TypedData<float> > data)
+{
+	MStatus s;
+	MAngle t;
+	t.setUnit(MAngle::kRadians);
+	t.setValue(data->readable());
+	s = plug.setValue(t);
+	return s;
+}
+
+} // namespace IECoreMaya
+
+
+template <typename DataType, typename UnitType>
+MStatus ToMayaPlugConverter::convertAttr(MPlug &plug) const
+{
+	MStatus s;
+	if (plug.isArray())
+	{
+		ConstObjectPtr toConvert = srcParameter()->getValidatedValue();
+		IECore::CompoundData::ConstPtr data = IECore::runTimeCast<const IECore::CompoundData>( toConvert );
 
+		typedef TypedData< std::vector<DataType> > VecDataType;
+		const VecDataType *indexedData = data->member<VecDataType>( "data", false );
+		const IntVectorData *indices = data->member<IntVectorData>( "indices", false );
+
+		// set all indexed values
+		assert( indexedData->readable().size() == indices->readable().size() );
+		for (int i=0; i<indexedData->readable().size(); ++i)
+		{
+			boost::intrusive_ptr<IECore::TypedData<DataType> > p = new IECore::TypedData<DataType>;
```

Comment:
Why do we need to construct a TypedData here? It seems to be thrown away as soon as we've set the plug value. Maybe that indicates we need a version of `setPlugValue` that takes raw data instead of TypedData.